scripts/pkg/build_templates/opensearch/deb/debian/preinst

```diff
@@ -23,6 +25,25 @@ if command -v systemctl >/dev/null && systemctl is-active opensearch-performance
     systemctl --no-reload stop opensearch-performance-analyzer.service
 fi
 
+# Check if OPENSEARCH_INITIAL_ADMIN_PASSWORD is defined
+# TODO:
+# 1. This check will need to be modified if there will be a min dist for deb in future (currently there is none)
+# 2. Currently, the demo config setup is defined to run, in postinst, if `opensearch-security` is present. Cannot apply the same check here since the plugins folder is not available yet.
+
+# Check if this is an upgrade by checking whether opensearch already exists
+OPENSEARCH_INSTALLED=$(dpkg -l | grep opensearch)
+
+OPENSEARCH_REQUIRED_VERSION="2.12.0"
+COMPARE_VERSION=`echo $OPENSEARCH_REQUIRED_VERSION $OPENSEARCH_VERSION | tr ' ' '\n' | sort -V | uniq | head -n 1`
+
+if [ -n "$OPENSEARCH_INSTALLED" ] && [ $OPENSEARCH_VERSION != CHANGE_VERSION ]; then
```

Comment:
This test feel bogus:

1. If `$OPENSEARCH_INSTALLED` is not empty, OS is already installed so I think we should assume passwords where previously seeded, when installing the first time, and should not be required anymore. Was `-z` (empty) intended in place of `-n` (not empty)?
2. `$OPENSEARCH_VERSION` is set to `CHANGE_VERSION` on line 14 (both or changed to a specific value by `scripts/components/OpenSearch/install.sh`), so the second part of the test is always true.

Reply:
1. -n was intended by I'm changing it to match exit codes.
2. CHANGE_VERSION is a placeholder and the value is being replaced with sed. See [here](https://github.com/opensearch-project/opensearch-build/pull/4332/files#diff-18196b41230cd78d55ad94c1f4b6b54ad7c225aaa48b16495f6d4352e26e760cR87)